services/py-api/src/service/participants_service.py

```diff
@@ -8,19 +8,22 @@
 from src.database.repository.participants_repository import ParticipantsRepository
 from src.database.repository.teams_repository import TeamsRepository
 from src.database.transaction_manager import TransactionManager
-from src.server.exception import DuplicateEmailError, DuplicateTeamNameError
+from src.server.exception import DuplicateEmailError, DuplicateTeamNameError, CapacityExceededError
 from src.server.schemas.request_schemas.schemas import ParticipantRequestBody
 
 
 class ParticipantService:
     """Service layer responsible for handling the business logic when registering a participant"""
 
     def __init__(
-        self, participant_repo: ParticipantsRepository, team_repo: TeamsRepository, tx_manager: TransactionManager
+        self, participant_repo: ParticipantsRepository, team_repo: TeamsRepository, tx_manager: TransactionManager,
+        hackathon_cap: int, team_capacity: int 
     ) -> None:
         self._participant_repo = participant_repo
         self._team_repo = team_repo
         self._tx_manager = tx_manager
+        self.hackathon_cap = hackathon_cap
+        self.team_capacity = team_capacity
```

Comment:
why are these added as constructor params? I think they should be extracted as constants. @Isak-Bego can also say.

Reply:
Fixed f33e8f7a85dbefbcda9a089a8ba5a1d3e7e93e9b